src/Queue/RabbitMQQueue.php

```diff
@@ -72,16 +72,35 @@ public function pushRaw($payload, $queueName = null, array $options = [])
              */
             [$queue, $topic] = $this->declareEverything($queueName);
 
+            /** @var AmqpMessage $message */
             $message = $this->context->createMessage($payload);
-            $message->setRoutingKey($queue->getQueueName());
+
             $message->setCorrelationId($this->getCorrelationId());
             $message->setContentType('application/json');
             $message->setDeliveryMode(AmqpMessage::DELIVERY_MODE_PERSISTENT);
 
+            if (isset($options['routing_key'])) {
+                $message->setRoutingKey($options['routing_key']);
+            } else {
+                $message->setRoutingKey($queue->getQueueName());
+            }
+
+            if (isset($options['priority'])) {
+                $message->setPriority($options['priority']);
+            }
+
             if (isset($options['expiration'])) {
                 $message->setExpiration($options['expiration']);
             }
 
+            if (isset($options['delivery_tag'])) {
+                $message->setDeliveryTag($options['delivery_tag']);
+            }
+
+            if (isset($options['consumer_tag'])) {
+                $message->setConsumerTag($options['consumer_tag']);
+            }
```

Comment:
What is the reason for setting consumer_tag and deliver_tag here? They make sense for the consumer but not producer.

Reply:
Also, they should not be set manually, the broker sets them

Reply:
well, since the message have a lot of configuration/property, I'd like to put them all to avoid other PRs, like my previous one
what do you think?

Reply:
A developer should not set delivery_tag or consumer_tag never. They are set by the broker only.

Reply:
fine for me

Reply:
Why these methods even exists then?